### ISSUE TITLE
potlatch creates invalid ways having only one node or multiple consecutive identical nodes

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#2501]
**[Submitted to the original trac issue database at 11.03am, Wednesday, 25th November 2009]**

* a way with only one node makes no sense: http://wiki.openstreetmap.org/wiki/Data_Primitives#Way

* a way can have identical nodes (e.g. circular ways), but there is no reason for having an identical node succeeding itself within a way.

please refer to the history of the ways in changeset http://www.openstreetmap.org/browse/changeset/3210744. in almost all cases potlatch was the editor in question generating those errors, before xybot repaired the errors.

**Comment by tomhughes** [comment]
**Author: Richard**
**[Added to the original trac issue at 1.33pm, Tuesday, 22nd December 2009]**

Steps to reproduce please or I'll close this.

**Comment by tomhughes** [comment]
**Author: xylome**
**[Added to the original trac issue at 1.23pm, Tuesday, 26th January 2010]**

this is a serious issue, please investigate this further!

i don't have the slightest idea how this could be reproduced (i'm using JOSM as editor), but there are hundreds of those ways produced each day.
see the changesset of xybot http://www.openstreetmap.org/user/xybot/edits "removing multiple consecutive identical nodes in ways/delete one node ways" as a reference!

**Comment by tomhughes** [comment]
**Author: Richard**
**[Added to the original trac issue at 1.48pm, Tuesday, 26th January 2010]**

Replying to [comment:2 xylome]:
> i don't have the slightest idea how this could be reproduced (i'm using JOSM as editor)

Then I would ask you to open Potlatch for once in your life, and try. I have a million and one people asking me to do things; if you want your particular itch scratched then you'll need to at least do a bit of the work yourself.

**Comment by tomhughes** [comment]
**Author: xylome**
**[Added to the original trac issue at 12.39pm, Thursday, 28th January 2010]**

let me first say, that i appreciate your work very much and that potlatch does a very good jobs in giving people a jumpstart into osm.

but it's not an itch of mine and yes i have no idea how this erroneous behaviour can be reproduced. i'm only seeing the results. it's a data integrity problem potlatch is producing here.

maybe you could do a validation of the ways potlatch produces before they get uploaded.

**Comment by tomhughes** [comment]
**Author: Richard**
**[Added to the original trac issue at 12.55pm, Thursday, 28th January 2010]**

That's kind of the problem, though - as far as I know (I haven't looked at that code for a while) Potlatch _does_ validate this. There's a method in way.as called removeDuplicates that makes sure that if, for example, you have a way ABCB and C is removed, one of the Bs will be removed too. This is why I'm keen to get a reproducible test case.

**Comment by tomhughes** [comment]
**Author: xylome**
**[Added to the original trac issue at 1.28pm, Thursday, 28th January 2010]**

for me it seems as if it happens, when people try to doubleclick to end drawing a way. most of the multiple identical nodes are on the end of a way and do not seem to be the result of a deletion of a node in the middle of a way.

**Comment by tomhughes** [comment]
**Author: xylome**
**[Added to the original trac issue at 1.32pm, Thursday, 28th January 2010]**

some of the ways are one-node-ways. it seems as if the user deletes one node of a two-node-way. the result should be, that also the affected way should be deleted then. the last remaing node should also be deleted if it is no member of another way or relation.

**Comment by tomhughes** [comment]
**Author: Richard**
**[Added to the original trac issue at 1.43pm, Thursday, 28th January 2010]**

The hint about it being the end of a way is really helpful - thank you. I'll have a look next week once I've got a bit more free time.

**Comment by tomhughes** [comment]
**Author: Ilis**
**[Added to the original trac issue at 3.56am, Friday, 29th January 2010]**

I have such errors using Potlatch. I dont no steps to reproduce, but it often happens when I have add point on crossing ways with Shift+Click.

**Comment by tomhughes** [comment]
**Author: katzlbt**
**[Added to the original trac issue at 11.59am, Sunday, 7th February 2010]**

I would bet it can be done by linking the way to itself and deleting the node before.

**Comment by tomhughes** [comment]
**Author: NE2**
**[Added to the original trac issue at 12.04pm, Monday, 8th February 2010]**

Is this possibly related to http://trac.openstreetmap.org/ticket/2700 ?

**Comment by tomhughes** [comment]
**Author: koenigl@fs.tum.de**
**[Added to the original trac issue at 1.43am, Sunday, 28th February 2010]**

Replying to [comment:1 Richard]:
> Steps to reproduce please or I'll close this.

Im not sure whether this is related to the problem discussed here (This is my second night with OSM ... ;-) but I just made the following observation in Potlatch:

After heavy load due to zooming out to far (during which the OS and Flash plugin raised message boxes concering non-reacting applications and Flash scripts, respectively) apparently keyboard input to Potlatch stopped working. At least, Potlatch did not seem to react to neither ESC, nor DEL, nor R. As a consequence, the only possibility (I knew about) to cancel the drawing of a new way that was started unintentionally was to click the first endpoint again, resulting in a single point (black-lined green circle in Potlatch) and maybe a way containing that point (that would have to be invisible). I left one of these near lat=48.06028, lon=11.47237, tagged fixme.

Behavior of Potlatch returned to normal after navigating (via the Browser) to map and then back to edit.

... hope that helps ...

**Comment by tomhughes** [comment]
**Author: Richard**
**[Added to the original trac issue at 3.33pm, Sunday, 28th February 2010]**

It's not related afaik.

**Comment by tomhughes** [comment]
**Author: Sanderd17**
**[Added to the original trac issue at 7.21am, Thursday, 15th April 2010]**

Sometimes, I see black dots. When you click on one, it always produces a street, even if you just want to select the dot. One of those dots are the traffic lights in http://www.openstreetmap.org/edit?lat=50.94854&lon=3.13093&zoom=17 . Hope that has something to do with the bug. If not, it's another bug.

Sander

**Comment by tomhughes** [comment]
**Author: Sanderd17**
**[Added to the original trac issue at 9.21am, Thursday, 15th April 2010]**

OK, all black dots are traffic lights, but I still don't get why these dots are treated in an other way than normal nodes.

**Comment by tomhughes** [comment]
**Author: skippern@gimnechiske.org**
**[Added to the original trac issue at 11.30am, Friday, 23rd April 2010]**

I had this bug occur doing import from scripts, it might be that this bug is not related to Potlatch at all. In my case it was DXF files converted to OSM XML and than split in layers. I discovered the connection between the scripts and multiple node bug when installing Validator plugin in JOSM. I have not checked if there are other activities behind the other occurrences of this bug. I have done my best to clean out these multiple nodes.

There are no one-node ways in my import as far as I can tell.

**Comment by tomhughes** [comment]
**Author: mcld**
**[Added to the original trac issue at 8.04pm, Tuesday, 18th May 2010]**

xybot is asking me to give you "hints about how the bug can be recreated" (from <http://www.openstreetmap.org/history?bbox=-3.46892%2C51.19971%2C-3.44732%2C51.20855>). I didn't know I was creating these problems (pretty new to osm) so I'm not sure what to say. I'm just manually adding paths on my mac laptop (firefox 3.6.3), although I'm using a graphics tablet to draw them so that may be the issue? e.g. some small movements while clicking? That's pretty much all I can say at the moment, but feel free to ask me questions.

**Comment by tomhughes** [comment]
**Author: NE2**
**[Added to the original trac issue at 5.16pm, Thursday, 20th May 2010]**

I created this node when joining two crossing ways (shift-click): http://www.openstreetmap.org/browse/node/740243448
It was in Arcade Place four times, but I fixed that. I kept the dupe in LaSalle as evidence. My workflow was to start at the west end of Arcade, joining it to each cross street in succession.

No, I don't have directions to reproduce. But if you go to http://www.openstreetmap.org/edit?lat=41.9022109&lon=-87.6717511&zoom=18&way=24095618 and join this way to each cross street in succession, you'll sometimes get this. It happens annoyingly infrequently.

**Comment by tomhughes** [comment]
**Author: mrpulley**
**[Added to the original trac issue at 5.34am, Thursday, 24th June 2010]**

Here's another one that I inadvertently caused:
Changeset: http://www.openstreetmap.org/browse/changeset/4905787
Way with multiple identical nodes: http://www.openstreetmap.org/browse/way/51191127

When I created the offending node (http://www.openstreetmap.org/browse/node/764758248 ), I was creating a new way (http://www.openstreetmap.org/browse/way/61134997 ). I can't remember if I created the new way, aligned it, then created the junction node (then removing the ends), or if I just clicked on the pre-existing way to create the node. Either way, the duplicate nodes only existed (until fixed by Xybot) on the pre-existing way. I did not split the way and then join the two ends up, so there is no obvious way for me to create the multiple consecutive nodes.

I was using Potlatch 1.4, MacOS 10.5, using Firefox. Can't remember the Flash version (I'm at work at the moment). Hope this helps.

**Comment by tomhughes** [comment]
**Author: mrpulley**
**[Added to the original trac issue at 10.49am, Friday, 25th June 2010]**

I'm using Firefox 3.6.3, flash version 10.0 r45.

**Comment by tomhughes** [comment]
**Author: phillipsjk**
**[Added to the original trac issue at 9.32am, Friday, 3rd September 2010]**

Replying to [comment:1 Richard]:
> Steps to reproduce please or I'll close this.

I noticed that when joining ways, the editor prompts you to click to join when the cursor is over the node of the old way. After clicking, the user is asked to repeat the process even if the ways are already joined at that node. After clicking a second time, the user is no longer prompted. 

I tried to reproduce the problem with http://www.openstreetmap.org/browse/changeset/5668901 (save, rather than live edit.) 
I also tried to reproduce the problem with a live edit session http://www.openstreetmap.org/browse/changeset/5668981
It appears the correct number of nodes were generated in both cases. 

Potlatch 1.4 (save en)
While I doubt the flash version matters, I was using Flash 10.1.53.64 installed as a plug-in for Mozilla/5.0 (X11; U; Linux i686; en-US; rv:1.9.0.3) Gecko/2008092510 Ubuntu/8.04 (hardy) Firefox/3.0.3

**Comment by tomhughes** [comment]
**Author: phillipsjk**
**[Added to the original trac issue at 4.25pm, Friday, 17th September 2010]**

By accident, if discovered that xybot cleaned up after me: http://www.openstreetmap.org/browse/way/46785224/history
Node in question: http://www.openstreetmap.org/browse/node/914731575

I created that node by sift-clicking where the two ways joined. I may have used Ctrl-click first, but that appeared to have no effect.

Attempt to reproduce by joining ends appear to have failed: http://www.openstreetmap.org/browse/changeset/5804821

Attempt to reproduce by joining nodes on a self-overlapping way failed (potlatch does not appear to support that.

Created two overlapping ways: http://www.openstreetmap.org/browse/changeset/5804908

Success!: Created two redundant nodes on lower way; I dragged the connection around to ensure the two ways were connected (should check if bug is present without dragging, but I won't). http://www.openstreetmap.org/browse/way/77908536 
In changeset: http://www.openstreetmap.org/browse/changeset/5804934

Summary: reproduced the "potlatch creates invalid ways having ... multiple consecutive nodes" part of bug.

**Comment by tomhughes** [comment]
**Author: Richard**
**[Added to the original trac issue at 4.35pm, Friday, 17th September 2010]**

Good news but I'm having awful trouble parsing any of those sentences! And I can't really see what those ways are because you've deleted them.

Could you provide simple instructions to reproduce, along the lines of "click to start a new way, draw some points, double-click to end" etc. etc.?

**Comment by tomhughes** [comment]
**Author: phillipsjk**
**[Added to the original trac issue at 11.33pm, Friday, 17th September 2010]**

__Steps to reproduce:__
 1. Create two overlapping ways with no common nodes.
  * Save.
 2. Select one way, then shift-click the intersection to join the two ways.
 3. (needs further investigation) Move newly created point to check ways really are joined
  * Save.
 4. Examine history of the 'lower' way. You should see some duplicated nodes.

*To view the Deleted way history:*
 1. Follow link showing way history
 2. Click on changeset
 3. Select appropriate way
 4. select "view history" link at bottom of page
 5. Resulting link: http://www.openstreetmap.org/browse/way/77908536/history

No, the deleted way is not rendered :P

**Comment by tomhughes** [comment]
**Author: Richard**
**[Added to the original trac issue at 8.44am, Saturday, 18th September 2010]**

Very strange. I can't reproduce that at all, either on my test box or the live site. The ways always come out perfectly formed when I follow those instructions, whether in save or live mode, with thin ways or thick ways, whatever I try.

(And yes I do know how to view deleted ways, obviously, it's just that I find using the data browser a PITA. :) )

**Comment by tomhughes** [comment]
**Author: phillipsjk**
**[Added to the original trac issue at 5.27pm, Saturday, 18th September 2010]**

Did you turn on extra debugging? when trying to reproduce the error again, I got an error message I couldn't read completely due to a character set I don't understand.

**Possible things to check:**

*B: Does moving the node have an effect?*
 * Test without moving; use history to verify connection
  * Joined ways by selecting 'top' way, then shift-clicking on intersection. Did not drag new node to confirm connection. -- Result appears to be correct -- http://www.openstreetmap.org/browse/changeset/5812647
 * Test with moving
  * Joined ways by selection 'top' way, then shift-clicking on intersection. Dragged new node around to confirm connection. Note: This time editor seemed to start a new way at intersection; I don't know what I did differently when creating new node.

***Server said:*** 
*(Copy&paste does not work with potlatch; UTF-8 would be so much easier)*

```An unusual error happened (in 'startchangeset').```

```The server said:```*(Chinese or Japanese characters follow)*

 {Stroke from center to lower left}{Short stroke from upper right to center;longer stroke from center to lower right}{vertical stroke on upper left; 'J' on opposing side}{medium horizontal stroke near top with uptick;long horizontal stroke slightly below first;Both preceding strokes top a 'J'}{long horizontal stroke}{two ticks in upper left; 'J' starting at right below half-way point}{Backward 'E'}{single tick in uper left; 'J' starting at right below half-way point}{Vertical stroke on left with tick at bottom; two short horizontal strokes on right}{Upside-down 'Y'; two medium horizontal strokes above center; short vertical stroke on left edge of horizontal strokes}{Man with ladder on left}{'L' with bottom portion up-turned}{Downstroke with clockwise curl at bottom; two medium horizontal strokes}{'L' with bottom portion up-turned}{Vertical stroke from left of center to bottom left; short horizontal stoke through downstroke; Short horizontal stoke center-right; medium horizontal stroke lower right}{small circle lower left}{white space}{'\'}{'/';Vertical stoke on right}{downstroke with clockwise curl at bottom; single medium horizontal stoke near top} 

*(newline)*

 {'2';'5';'5'}{man}{'7' wearing flat, broad hat; long horizontal stoke at center}{downstroke with uptick on left; floating tick upper left; upside-down 'Y' on right}{small up-side-down 'Y', set inside upstroke-horizontal stroke-downstroke}{Short horizonatal stroke upper left;'C';tick just above center-right}{Lambda symbol}{'J' on left; horizontal stroke-downstroke with tick}{'L' with uptutned bottom}{top horizontal stoke-'C'}{'<'}{vertical stoke from top left of center to bottom left; short horizontal stroke through top of downstroke; 3 short horizontal strokes on right}{vertical stroke ending in small 'c'; medium horizontal stroke above-center}{meduim vertical stroke with uptick top-left;tick on right above center}{small circle lower left}

*(newline)*

```Please try again: Potlatch will start a new```

```changeset.```

``````

```Please e-mail richard@(...) with a bug```

```report, saying what you were doing at the time.```

***--- End server message ----***

"Tainted" changeset: http://www.openstreetmap.org/browse/changeset/5813172

*A: does it matter which way is selected before adding a node?*
 * Shift-click without selecting either way
 * Shift-click selecting the bottom way
 * Shift-click selecting the top way

*C: Does the flash version have an effect?*
 *  I have a copy of flash 9 I can use
 * May be able to test with a more up-to date browser

*D: Do time constraints have an effect?*
 * Does this only affect older computers with the CPU time maxed out?

**Comment by tomhughes** [comment]
**Author: phillipsjk**
**[Added to the original trac issue at 5.31pm, Saturday, 18th September 2010]**

Oops: Forgot to mention: I am in the habit of [C]losing the changeset before saving. This is an attempt to force the comment dialog box to appear.

**Comment by tomhughes** [comment]
**Author: Richard**
**[Added to the original trac issue at 11.12am, Sunday, 19th September 2010]**

No, I haven't changed anything.

**Comment by tomhughes** [comment]
**Author: phillipsjk**
**[Added to the original trac issue at 2.15am, Tuesday, 21st September 2010]**

I am going to have to echo 'NE2' and says I can no longer reliably reproduce the bug. My latest attempt to follow my reproduction steps yielded correct results as far as I can tell: http://www.openstreetmap.org/browse/changeset/5833987

I have noticed that my mouse does not always debounce properly under load, but I was unable to induce errors by triple-clicking, or by running Prime95 in background with default blended test settings. Firefox was using hardly any CPU time at all, even with Prime95 +19 niced (maximum; lowest priority). I had to stop prime95 to get the editor to draw the map (maybe I wasn't patient enough; but interrupts (disk paging) were using 0% of the CPU at the time).

I have confirmed the CPU and memory appear to be working correctly. I ran Prime95 with blended test, but reduced memory footprint (230MB instead of 760MB) for over 24 hours. No errors were reported. Potlatch was able to edit normally without disk paging. Because the mersenne.org download page was not working, I had to get the windows version from a mirror (ran it under wine).

**Comment by tomhughes** [comment]
**Author: phillipsjk**
**[Added to the original trac issue at 2.27am, Tuesday, 21st September 2010]**

note: just checked the 'top' man page: what I was calling 'interrupts' was actually 'iowait'; "Amount of time the CPU has been waiting for I/O to complete." (probably DMA transfers). With DMA enabled, interrupts rarely use more than a few % of CPU time.

I suppose it is possible the 'niced' prime95 was using 90% of the CPU time because firefox was spending 90% of the time waiting on the disk :P
Prime95 didn't have to wait because it was in memory already.

**Comment by tomhughes** [comment]
**Author: phillipsjk**
**[Added to the original trac issue at 1.43am, Monday, 27th September 2010]**

I again failed to reproduce the problem with my reproduction steps. (http://www.openstreetmap.org/browse/changeset/5888031)

The only thing I can think of that I changed was the deletion of Flash cookies. I am using a linux version of flash that allows you to drop it into the mozilla pluging directory of a single user without affecting the whole system. The files I deleted are stored in:

```~/.macromedia/Flash_Player/#SharedObjects/RandomString/www.openstreetmap.org/potlatch/potlatch.swf/preferences.sol```

```~/.macromedia/Flash_Player/macromedia.com/support/flashplayer/sys/#www.openstreetmap.org/settings.sol```

```~/.macromedia/Flash_Player/macromedia.com/support/flashplayer/sys/settings.sol```


I may need to resort to white-box debugging to track this down. I have never coded in flash in part because I oppose it in principle. I am using Potlatch instead of JOSM because I am lazy (still don't have my workstation "just so"; don't want to lose track of 20GB.).

The original report says that Potlach was the editor most commonly implicated. Which editor is most popular? Could the problem be server-side, but Potlatch gets implicated because it interacts with the server more often?

**Comment by tomhughes** [comment]
**Author: ghia**
**[Added to the original trac issue at 4.14am, Wednesday, 17th November 2010]**

I analysed a changeset
http://www.openstreetmap.org/browse/changeset/6386716
 on Zeezwaluwstraat
http://www.openstreetmap.org/browse/way/7497269
Before it was a simple crossing with the Meeuwenstraat in line with the other part of Zeezwaluwstraat
http://www.openstreetmap.org/browse/way/7496892
(See not yet updated map eg bicycle)
The first part was moved to the right. This means that a new point has to be created.
http://www.openstreetmap.org/browse/node/991638759
This did happen. 
Next this point should be reconnected and be part of the Meeuwenstraat.
http://www.openstreetmap.org/browse/way/7496787
In stead the point got duplicated and added again to the Zeezwaluwenstraat, where it already belonged to.
http://www.openstreetmap.org/browse/way/7497269/history (see v3) 
Not sure which operations you would need in Potlach to do such an operation.

Other case is 
http://www.openstreetmap.org/browse/changeset/6386246
There a point is deleted
http://www.openstreetmap.org/browse/node/412283516/history
It was between two streets
Rue Ren Binet
http://www.openstreetmap.org/browse/way/85512773
and Route de Sens 
http://www.openstreetmap.org/browse/way/35149034
The new midpoint became
http://www.openstreetmap.org/browse/node/412283517
of Route de Sens, which became then doubly added to the other road Rue Ren Binet.

Weird is also that in this changeset several version of the same object are present.

**Comment by tomhughes** [comment]
**Author: leighghunt@gmail.com**
**[Added to the original trac issue at 8.59pm, Thursday, 18th November 2010]**

Hi - in OSM changeset 6379862 for way 85321659 (http://www.openstreetmap.org/browse/way/85321659/history) there was the following comment:

> removing multiple consecutive identical nodes in ways/delete one node ways. please help by giving hints, about how this potlatch bug can be reproduced http://trac.openstreetmap.org/ticket/2501

I edited these polygons first in Potlatch as three polygons, and then merged them into one polygon in JOSM, so this was probably user error, rather than a bug in Potlatch.

Cheers,
Leigh.

**Comment by tomhughes** [comment]
**Author: nakor.wp@gmail.com**
**[Added to the original trac issue at 9.03pm, Monday, 6th December 2010]**

This is very easy to reproduce in Potlatch 2: select a node at the end of a way, then double-click on it -> the node is now duplicated in the way.

In Potlatch 1, use the same process but it only "works"if you double click on a node after it turned blue.

See changeset http://www.openstreetmap.org/browse/changeset/6568527 (P2) and http://www.openstreetmap.org/browse/changeset/6568770 (P1) where I was able to do it several times.

**Comment by tomhughes** [comment]
**Author: sdoerr**
**[Added to the original trac issue at 2.05pm, Friday, 8th April 2011]**

So could a developer perhaps 'accept' this supposedly 'new' bug report and see if they can fix it?

**Comment by tomhughes** [comment]
**Author: cucafera**
**[Added to the original trac issue at 8.21pm, Friday, 22nd April 2011]**

Changeset: http://www.openstreetmap.org/browse/changeset/7929894
While splitting the train track branches down here: http://www.openstreetmap.org/browse/way/91740051
Done to create a public transport relation:
http://www.openstreetmap.org/browse/relation/1542805

Hope it helps.
Tkanks a lot for your efforts, guys.

Alfonso.

**Comment by tomhughes** [comment]
**Author: bwnz**
**[Added to the original trac issue at 4.17am, Sunday, 8th January 2012]**


Yes, using Potlatch. Just had a look at history on a portion of map I edited 3 Jan (#10278715). Pretty sure no other changes have been made anywhere near here since BUT multiple instances of xybot claiming to be removing multiple consecutive identical nodes. Either message is indicating checking for problem rather than fixing, or problem may be introduced in data storage or manipulation after changeset saved. Which?

**Comment by tomhughes** [comment]
**Author: Nick Austin**
**[Added to the original trac issue at 10.29pm, Tuesday, 28th February 2012]**

I've actully seen this bug several times while editing. In all cases I was just editing normally - adding points to the end of a way. The same as I've done many times previously and many times since without the bug appearing. In order words I don't have a sequence that reproduces the bug because it's just something that happens randomly.

However on several occasions I was able to use Z to undo the problem and this has given clues as to what happened.  I actually see two variants of this problem one of which xybot can catch and the other just results in a very odd shapped way. I'll explain both as best as I can in case either is of help.

Variant 1:
Having added several points to a way I click on a blank area of the map. A new node appears at the cursor but instead of joining the new point to the previous point Potlatch joins it to a completely different point somewhere in middle of the current way. If I press Z (undo) the newly created node disappears but the line drawn from the cursor now goes to the node in the middle of the way. If I press Z again I'm back where I was before I the mouse click. The way I see it a single mouse-click has for some reason resulted in Potlatch attempting to add two points to the line, the first of which is a duplicate of a point already along the line and the second point at the cursor. Hope that makes sense.

Variant 2:
Very similar. Having added several points to a way I click on a blank area of the map. A new node appears at the cursor but instead of joining the new point to the previous point Potlatch joins it to some random point several km away from the current cursor location. If I don't notice and continue editing it results in a way which has what I can only describe as a spike. As with variant 1 it takes two presses of Z to delete the spike and put things back to normal.

**Comment by tomhughes** [comment]
**Author: bill42**
**[Added to the original trac issue at 10.48pm, Sunday, 11th March 2012]**

I just noticed this on a way I edited yesterday:
http://www.openstreetmap.org/browse/way/6377054

I remember joining nodes that were consecutive on the street.

Steps to reproduce:
Have 3 ways, lets call the A B and C.
Make B a vertical way with 4 nodes.
Make A a horizontal way connecting to the 2nd node of B.
Make C a horizontal way connecting to the 3rd node of B.
(Save)
Select A
Move 2nd node of B to over 3rd node of B.
Use J to join.
(Save)
B still has 4 nodes, with the 2nd and 3rd nodes being duplicated.

Here is the Way "B" from my attempt to reproduce.
http://www.openstreetmap.org/browse/way/154594186/history

**Comment by tomhughes** [comment]
**Author: Oli-Wan**
**[Added to the original trac issue at 2.16pm, Thursday, 11th April 2013]**

Some more material:

http://www.openstreetmap.org/browse/way/215524538       v1      2013-04-05      Potlatch 2
http://www.openstreetmap.org/browse/way/213101350       v1      2013-03-27      Potlatch 2
http://www.openstreetmap.org/browse/way/212453827       v6      2013-03-30      Potlatch 2
http://www.openstreetmap.org/browse/way/211359362       v4      2013-03-22      Potlatch 2
http://www.openstreetmap.org/browse/way/211355851       v5      2013-03-22      Potlatch 2
http://www.openstreetmap.org/browse/way/211100413       v2      2013-03-28      Potlatch 2
http://www.openstreetmap.org/browse/way/210965495       v1      2013-03-19      Potlatch 2
http://www.openstreetmap.org/browse/way/210965264       v1      2013-03-19      Potlatch 2
http://www.openstreetmap.org/browse/way/209806652       v5      2013-03-30      Potlatch 2
http://www.openstreetmap.org/browse/way/208247380       v3      2013-03-04      Potlatch 2
http://www.openstreetmap.org/browse/way/205206803       v1      2013-02-12      Potlatch 2
http://www.openstreetmap.org/browse/way/205206802       v1      2013-02-12      Potlatch 2
http://www.openstreetmap.org/browse/way/204058150       v1      2013-02-03      Potlatch 2
http://www.openstreetmap.org/browse/way/204058149       v1      2013-02-03      Potlatch 2
http://www.openstreetmap.org/browse/way/204058142       v1      2013-02-03      Potlatch 2
http://www.openstreetmap.org/browse/way/202790640       v1      2013-01-25      Potlatch 2
http://www.openstreetmap.org/browse/way/199773822       v1      2013-01-06      Potlatch 2
http://www.openstreetmap.org/browse/way/195726321       v3      2013-01-12      Potlatch 2
http://www.openstreetmap.org/browse/way/193943222       v1      2012-12-01      Potlatch 2
http://www.openstreetmap.org/browse/way/191906898       v1      2012-11-21      Potlatch 2
http://www.openstreetmap.org/browse/way/187183024       v1      2012-10-23      Potlatch 2
http://www.openstreetmap.org/browse/way/184399078       v1      2012-10-06      Potlatch 2
http://www.openstreetmap.org/browse/way/184324574       v1      2012-10-05      Potlatch 2
http://www.openstreetmap.org/browse/way/184324182       v1      2012-10-05      Potlatch 2
http://www.openstreetmap.org/browse/way/184281194       v2      2013-03-07      Potlatch 2
http://www.openstreetmap.org/browse/way/184147952       v1      2012-10-04      Potlatch 2
http://www.openstreetmap.org/browse/way/184133317       v1      2012-10-04      Potlatch 2
http://www.openstreetmap.org/browse/way/179864106       v7      2013-03-02      Potlatch 2
http://www.openstreetmap.org/browse/way/172903364       v4      2013-03-15      Potlatch 2
http://www.openstreetmap.org/browse/way/148862458       v17     2013-03-25      Potlatch 2
http://www.openstreetmap.org/browse/way/143975448       v2      2012-10-05      Potlatch 2
http://www.openstreetmap.org/browse/way/95933816        v5      2013-02-16      Potlatch 2
http://www.openstreetmap.org/browse/way/95392017        v2      2013-02-21      Potlatch 2
http://www.openstreetmap.org/browse/way/82623990        v3      2013-03-26      Potlatch 2
http://www.openstreetmap.org/browse/way/49581948        v7      2013-03-26      Potlatch 2
http://www.openstreetmap.org/browse/way/37866489        v3      2012-10-04      Potlatch 2
http://www.openstreetmap.org/browse/way/25971080        v3      2013-01-24      Potlatch 2
http://www.openstreetmap.org/browse/way/20866575        v22     2013-02-24      Potlatch 2
http://www.openstreetmap.org/browse/way/7610771 v3      2013-02-28      Potlatch 2

http://www.openstreetmap.org/browse/way/216176015       v2      2013-04-07      Potlatch 1.4 (live de)
http://www.openstreetmap.org/browse/way/215268270       v1      2013-04-03      Potlatch 1.4 (save en)
http://www.openstreetmap.org/browse/way/210241655       v1      2013-03-15      Potlatch 1.4 (live en)
http://www.openstreetmap.org/browse/way/210241224       v1      2013-03-15      Potlatch 1.4 (live en)
http://www.openstreetmap.org/browse/way/209301534       v1      2013-03-11      Potlatch 1.4 (save fi)
http://www.openstreetmap.org/browse/way/209301533       v1      2013-03-11      Potlatch 1.4 (save fi)
http://www.openstreetmap.org/browse/way/206247773       v1      2013-02-19      Potlatch 1.4 (save de)
http://www.openstreetmap.org/browse/way/201921742       v1      2013-01-19      Potlatch 1.4 (live de)
http://www.openstreetmap.org/browse/way/201828067       v1      2013-01-18      Potlatch 1.4 (save fi)
http://www.openstreetmap.org/browse/way/200523494       v1      2013-01-11      Potlatch 1.4 (save fi)
http://www.openstreetmap.org/browse/way/198043826       v1      2012-12-25      Potlatch 1.4 (live de)
http://www.openstreetmap.org/browse/way/185796957       v2      2012-12-21      Potlatch 1.4 (save ru)
http://www.openstreetmap.org/browse/way/185774734       v1      2012-10-14      Potlatch 1.4 (save ru)
http://www.openstreetmap.org/browse/way/185714451       v1      2012-10-13      Potlatch 1.4 (live de)
http://www.openstreetmap.org/browse/way/146975547       v2      2013-01-20      Potlatch 1.4 (save en)
http://www.openstreetmap.org/browse/way/144132177       v4      2013-02-18      Potlatch 1.4 (save de)
http://www.openstreetmap.org/browse/way/140372276       v14     2013-03-01      Potlatch 1.4 (live de)
http://www.openstreetmap.org/browse/way/56304231        v17     2013-01-03      Potlatch 1.4 (save ru)
http://www.openstreetmap.org/browse/way/34641923        v9      2013-02-18      Potlatch 1.4 (save de)
http://www.openstreetmap.org/browse/way/31360752        v5      2013-02-18      Potlatch 1.4 (save de)
http://www.openstreetmap.org/browse/way/28320345        v6      2013-02-27      Potlatch 1.4 (save fi)

**Comment by tomhughes** [comment]
**Author: iandees**
**[Added to the original trac issue at 7.52pm, Monday, 9th September 2013]**

Cleaning aging tickets. If someone can nail down a reproducible problem feel free to reopen this in the potlatch2 component.
